odxtools/basicstructure.py

```diff
@@ -287,10 +287,13 @@ def _message_format_lines(self, allow_unknown_lengths: bool = False) -> List[str
         # sort parameters
         sorted_params: list = list(self.parameters)  # copy list
 
-        def param_sort_key(param: Parameter) -> Tuple[int, int]:
+        def param_sort_key(param: Parameter) -> Tuple[int, int, int]:
```

Comment:
@andlaus why are parameters being sorted here? does the spec indicate that as a pre-condition?